### PR TITLE
Skip softmax calibration via Triton kernel

### PR DESCRIPTION
**Pull request by rohansjoshi**
### What does this PR do?

Adds skip softmax calibration for LLMs via Triton kernel (leveraging existing kernel used for diffusion)

Type of change: New feature

### Usage

```
python hf_sa.py --pyt_ckpt_path Qwen/Qwen3-8B --sparse_attn skip_softmax_triton_calib
```

The Triton calibration equals PyTorch at every threshold, for both phases:
| threshold  | prefill triton/pytorch | decode triton/pytorch |
|------|------------------------|-----------------------|
| 0.30 | 0.0% / 0.0%            | 12.5% / 12.5%         |
| 0.50 | 0.0% / 0.0%            | 37.5% / 37.5%         |
| 0.70 | 10.0% / 10.0%          | 62.5% / 62.5%         |

### Before your PR is "*Ready for review*"

Make sure you read and follow [Contributor guidelines](https://github.com/NVIDIA/Model-Optimizer/blob/main/CONTRIBUTING.md) and your commits are signed (`git commit -s -S`).

Make sure you read and follow the [Security Best Practices](https://github.com/NVIDIA/Model-Optimizer/blob/main/SECURITY.md#security-coding-practices-for-contributors) (e.g. avoiding hardcoded `trust_remote_code=True`, `torch.load(..., weights_only=False)`, `pickle`, etc.).

- Is this change backward compatible?: ✅ / ❌ / N/A 
- If you copied code from any other sources or added a new PIP dependency, did you follow guidance in `CONTRIBUTING.md`: ✅ / ❌ / N/A 
- Did you write any new necessary tests?: ✅ / ❌ / N/A 
- Did you update [Changelog](https://github.com/NVIDIA/Model-Optimizer/blob/main/CHANGELOG.rst)?: ✅ / ❌ / N/A 
- Did you get Claude approval on this PR?: ✅ / ❌ / N/A 

## Summary by CodeRabbit

* **New Features**
  * Added a Triton-based skip-softmax sparse-attention calibration option and a CLI flag to override the calibration data directory (defaults to adjacent RULER data).

* **Bug Fixes**
  * Ensure calibration kernels run on the correct CUDA device; align measurement granularity and tile/block sizing; ignore padded query rows when counting skippable tiles.

* **Tests**
  * Added GPU Triton calibration tests for end-to-end inference, multi-threshold stats, and decode-phase reporting.

* **Documentation**
  * Updated changelog and example to expose the new option and flag.

### CODE REVIEW

File: modelopt/torch/kernels/common/attention/hf_triton_attention.py [hunk 1]
```diff
@@ -22,11 +22,71 @@
...
 
+# ---------------------------------------------------------------------------
+# Thread-local skip-softmax calibration config for the HF (modelopt_triton) backend
+# ---------------------------------------------------------------------------
+# Mirrors the diffusers/LTX backends: during calibration the Triton calibration
+# kernel measures multi-threshold tile-skip statistics without skipping any tiles.
+# Inference-time config (skip threshold / scale factor) is still read from the
+# module/method attributes in ``triton_attention_forward`` — only calibration
+# state lives here.
+_thread_local = threading.local()
```
Comment (kaix-nv) [inline modelopt/torch/kernels/common/attention/hf_triton_attention.py]: Can we remove `_thread_local`, `set_hf_triton_skip_softmax_config`, `clear_hf_triton_skip_softmax_config`?

The inference branch already reads `module._sparse_method_instance`, so this data path is proven. The `thread-local` just duplicates the method instance.

File: modelopt/torch/sparsity/attention_sparsity/config.py [hunk 2]
```diff
@@ -546,6 +546,36 @@ class FlashSkipSoftmaxConfig(SparseAttentionConfig):
...
 
+# RULER calibration via the fused Triton calibration kernel (prefill only).
+# Computes the same exponential-model calibration as SKIP_SOFTMAX_CALIB but
+# measures tile-skip statistics with the Triton ``attention_calibrate`` kernel
+# (the way the Triton inference kernel actually skips tiles) instead of the
+# PyTorch F.softmax-patching block path. Faster on GPU since it avoids
+# materializing per-block tensors.
+SKIP_SOFTMAX_TRITON_CALIB = {
+    "sparse_cfg": {
+        "calibration": {
```
Comment (kaix-nv) [inline modelopt/torch/sparsity/attention_sparsity/config.py:567]: We should support calibration for both prefill and decode

**Review by kaix-nv — APPROVED** [review]

**Review by shengliangxu — APPROVED** [review]
